### ISSUE TITLE
Hot reload is not working when using custom babel config

### ISSUE THREAD

**Issue by steverandy** [gatsbyjs/gatsby#30017]
## Description

When custom babel config is added to the project root, hot reload stops working.
It also might be related to importing specific package.

I managed to reproduce this issue when using:
`import store from "store"`

It could also be that I'm doing the config wrong. If that's the case, please help with correcting.

### Steps to reproduce

1. Clone https://github.com/steverandy/my-default-starter
2. Run develop mode.
3. Make text changes in `pages/page-2.js`

### Expected result

Page should reload and seeing new text.

### Actual result

Page does not reload and not seeing changes.

Browser console log:
```
[HMR] bundle rebuilding
client.js:250 [HMR] bundle rebuilt in 201ms
process-update.js:51 [HMR] Checking for updates on the server...
process-update.js:125 [HMR] Updated modules:
process-update.js:127 [HMR]  - ./src/pages/page-2.js
process-update.js:127 [HMR]  - ./.cache/_this_is_virtual_fs_path_/$virtual/async-requires.js
process-update.js:132 [HMR] App is up to date.
```

### Environment

```
System:
    OS: macOS 11.2.2
    CPU: (6) x64 Intel(R) Core(TM) i5-8500 CPU @ 3.00GHz
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 14.13.1 - /usr/local/bin/node
    Yarn: 1.22.10 - /usr/local/bin/yarn
    npm: 7.6.0 - /usr/local/bin/npm
  Languages:
    Python: 2.7.17 - /usr/local/bin/python
  Browsers:
    Chrome: 89.0.4389.72
    Safari: 14.0.3
  npmPackages:
    gatsby: ^3.0.1 => 3.0.1
    gatsby-plugin-gatsby-cloud: ^2.0.0 => 2.0.0
    gatsby-plugin-image: ^1.0.0 => 1.0.0
    gatsby-plugin-manifest: ^3.0.0 => 3.0.0
    gatsby-plugin-offline: ^4.0.0 => 4.0.0
    gatsby-plugin-react-helmet: ^4.0.0 => 4.0.0
    gatsby-plugin-sharp: ^3.0.0 => 3.0.0
    gatsby-source-filesystem: ^3.0.0 => 3.0.0
    gatsby-transformer-sharp: ^3.0.0 => 3.0.0
  npmGlobalPackages:
    gatsby-cli: 2.19.1
```

**Comment by vladar** [comment]
I could reproduce this. Thanks for the repro! 👍 Should be fixed via #30047

**Comment by vladar** [comment]
Published in `gatsby@3.0.3`
